### PR TITLE
ICDS Dashboard: Angularjs upgrade from 1.4.4 to 1.7.9

### PR DESCRIPTION
**Pull request by saiprakash-v**
https://app.asana.com/0/1185205808600689/1186445439810554

### CODE REVIEW

**Comment by saiprakash-v** [comment]
@Rohit25negi I didnot rename "directive.js" into "component.js", as git was considering directive file was deleted and new component file was added (because of this, actual changes are not highlighted)

File: custom/icds_reports/static/icds_reports/js/spec/adolescent-girls.directive.spec.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ describe('Adolescent Girls Directive', function () {
 
     beforeEach(module('icdsApp', function ($provide) {
         utils.provideDefaultConstants($provide, {});
-        $provide.constant("haveAccessToFeatures", false);
```
Comment (Rohit25negi) [inline custom/icds_reports/static/icds_reports/js/spec/adolescent-girls.directive.spec.js:18]: is this removed just because it was not used anywhere or its removed because way of injecting vars changed?

File: custom/icds_reports/static/js/angular-services/baseControllers.service.js [hunk 2]
```diff
@@ -4,6 +4,10 @@ window.angular.module('icdsApp').factory('baseControllersService', ['$timeout',
     var BaseFilterController = function ($scope, $routeParams, $location, dateHelperService, storageService,
         navigationService) {
         var vm = this;
+
+        $scope.$on('selected_locations_changed', function (event, data) {
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/angular-services/baseControllers.service.js:8]: why is it added here? and below in the same file

File: custom/icds_reports/static/js/directives/access-denied/access-denied.directive.js [hunk 3]
```diff
@@ -1,9 +1,6 @@
...
-var url = hqImport('hqwebapp/js/initial_page_data').reverse;
-
-window.angular.module('icdsApp').directive('accessDenied', function () {
-    return {
-        restrict: 'E',
-        templateUrl: url('icds-ng-template', 'access-denied.directive'),
-    };
+window.angular.module('icdsApp').component('accessDenied', {
+    templateUrl: function () {
+        var url = hqImport('hqwebapp/js/initial_page_data').reverse;
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/access-denied/access-denied.directive.js:3]: was there a reason for shifting the `url` to the local scope?

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js [hunk 4]
```diff
@@ -58,15 +58,17 @@ function AdditionalFilterController($scope, $location, $uibModal, storageService
...
-    // eg:vm.filters = ['gender', 'age']
-    // this array has filters that are not to be shown. so if 'gender' is not in array, it can be shown.
-    if (vm.filters && vm.filters.indexOf('gender') === -1) {
-        vm.showGenderFilter = true;
-        filtersObjects.push({ label: 'Gender', value: vm.selectedGender });
-    }
-    if (vm.filters && vm.filters.indexOf('age') === -1) {
-        vm.showAgeFilter = true;
-        filtersObjects.push({ label: 'Age', value: vm.selectedAge });
+    vm.$onInit = function () {
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js:61]: life cycle methods?

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js [hunk 5]
```diff
@@ -152,19 +154,14 @@ function AdditionalFilterController($scope, $location, $uibModal, storageService
 AdditionalFilterController.$inject = ['$scope', '$location', '$uibModal', 'storageService', 'genders', 'ages'];
 AdditionalModalController.$inject = ['$location', '$uibModalInstance', 'filters', 'genders', 'ages', 'agesServiceDeliveryDashboard', 'haveAccessToFeatures'];
 
-window.angular.module('icdsApp').directive("additionalFilter", ['templateProviderService', function (templateProviderService) {
-    var url = hqImport('hqwebapp/js/initial_page_data').reverse;
-    return {
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js:157]: some variables are removed from the return. is that related to upgrade or a cleanup?

File: package.json [hunk 6]
```diff
@@ -10,20 +10,20 @@
...
+        "angular-message-format": "1.7.9",
+        "angular-messages": "1.7.9",
         "angular-nvd3": "1.0.9",
         "angular-perfect-scrollbar": "0.2.0",
-        "angular-resource": "1.4.4",
-        "angular-route": "1.4.4",
-        "angular-sanitize": "1.4.4",
+        "angular-resource": "1.7.9",
+        "angular-route": "1.7.9",
+        "angular-sanitize": "1.7.9",
```
Comment (Rohit25negi) [inline package.json:26]: thanks for ensuring other libraries are also updated.

File: yarn.lock [hunk 7]
```diff
@@ -198,20 +198,20 @@ amdefine@>=0.0.4:
   resolved "https://registry.yarnpkg.com/amdefine/-/amdefine-1.0.1.tgz#4a5282ac164729e93619bcfd3ad151f817ce91f5"
   integrity sha1-SlKCrBZHKek2Gbz9OtFR+BfOkfU=
 
-angular-animate@1.4.4:
-  version "1.4.4"
-  resolved "https://registry.yarnpkg.com/angular-animate/-/angular-animate-1.4.4.tgz#70baec78ff49cba3be13ea0141819ffb2e785115"
-  integrity sha1-cLrseP9Jy6O+E+oBQYGf+y54URU=
+angular-animate@1.7.9:
```
Comment (Rohit25negi) [inline yarn.lock:201]: weird that lock file is also a part of git history.

**Review by Rohit25negi — COMMENTED** [review]
Overall looks good, mostly similar change of directive to the component. Still left a few comments. Also could you please create a copy PR here as well https://github.com/dimagi/commcare-icds so that we can do deploys

File: custom/icds_reports/static/icds_reports/js/spec/adolescent-girls.directive.spec.js (hunk 1 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/icds_reports/js/spec/adolescent-girls.directive.spec.js:18]: this is redundant, as the utils function in the line above it `provideDefaultConstants` add this constant

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js (hunk 4 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js:61]: yes. Accessing/changing the bindings of the component only  after they are initialized

File: custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js (hunk 5 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js:157]: its part of upgrade. Components doesn't need `restrict`, `bindToController`

File: yarn.lock (hunk 7 above)
Comment (saiprakash-v) [reply yarn.lock:201]: During the build process, `yarn install` is run with `frozen-version` flag. This must have something to do with this

File: custom/icds_reports/static/js/directives/access-denied/access-denied.directive.js (hunk 3 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/access-denied/access-denied.directive.js:3]: Yeah. One of the reasons was lint warnings of giving this variable redundant global scope. (I dont remember the other reason exactly :expressionless: )

**Comment by Rohit25negi** [comment]
Closing this Pr as this change is taken forward in [this](https://github.com/dimagi/commcare-hq/pull/28353/files) and in [this](https://github.com/dimagi/commcare-icds/pull/13/files)
